RELEASE.md

```diff
@@ -1,66 +1,79 @@
 
-Due to either github dependencies or a crate failing the build stage during publish not all crates are being published.
+Whenever a `PATCH` is introduced, it is applied to all the latest `MAJOR` releases.
+For example: imagine there's releases `v1.0.0`, `v1.1.0`, and `v2.0.0`. A bug is found, dating back all the way to `v1.0.0`.
+A patch is applied such that the following new tags are introduced: `v1.1.1` and `v2.0.1`.
 
-- [x] buffer_sv2
-- [x] binary_codec_sv2
-- [x] binary_sv2
-- [x] common_messages_sv2
-- [x] const_sv2
-- [x] derive_codec_sv2
-- [x] framing_sv2
-- [x] serde_sv2
-- [x] sv1_api
-- [] noise_sv2 - github depependency (ed25519-dalek)
-- [] codec_sv2 - noise dependency
-- [] template_distribution_sv2 - build failure
-- [] mining_sv2 - publishes after template_distribution_sv2
-- [] job-declaration_sv2 - publishes after template_distribution_sv2
-- [] roles_logic_sv2 - publishes after template_distribution_sv2
-- [] network_helpers - publishes after template_distribution_sv2
-- [] error_handling - had to exclude to bypass rate limit but will probably publish now
-- [] sv2_ffi - had to exclude to bypass rate limit but will probably publish now
-- [] all roles - noise dependency
+# Git Branching Strategy
 
-## Github Roles Releases
+We follow a simplified [gitflow](https://nvie.com/posts/a-successful-git-branching-model/) branching strategy.
 
-The roles binaries are released in github in all jobs other than the `crates_publish` job. To be able to publish,
-the job must be able to find a new tag (one that hasnt already been released), matching the crate name of the role to be
-released. For example, imagine there is currently a release for pool_sv2-v1.0.0. For the jobs releasing a new pool version to succeed,
-it must be able to find a tag containing the string "pool_sv2" with a version greater than 1.0.0. This should not be an issue since the tags
-are automatically created by the `autoversion.yaml` workflow, and moved into main when the generated PR is merged.
+Although our strategy is very similar to the classic gitflow model, we do not keep release branches.
 
-### Github Release Issue
+![](git-branching.png)
 
-Currently we do not support windows releases because we are unable to run the step to return the last tag for a given crate name. See the section above
-for an explaination of the process.
+## Principal Branches
 
+The SRI repo holds two principal branches with an infinite lifetime:
+- `main`
+- `dev`
 
+We consider `main` to be the branch where the source code of `HEAD` always reflects a production-ready state.
 
-## Github Binary Releases
+We consider `dev` to be the branch where the source code of `HEAD` always reflects a state with the latest delivered development changes for the next release.
 
-# Versioning
+Every 4 weeks all the changes to `dev` are merged back into `main` and then tagged with a release number while bumping `MAJOR` and/or `MINOR`.
```

Comment:
I agree with all @pavlenex's considerations about the 4 weeks period. I'm very skeptical about it and I don't see a need for it to be that strictly to be honest. Why not defining a time period at all? Or put it larger (2 months)? 
Since with this release we finally implemented everything from the specs, I don't expect that many new features in a short-mid time frame. I'm sure we will need to make more patch releases than feature ones btw

Reply:
ok this point is generating a lot of confusion so I propose we leave the document like this for now:
```diff
- Every 4 weeks all the changes to `dev` are merged back into `main` and then tagged with a release number while bumping `MAJOR` and/or `MINOR`.
+ The SRI team will decide the appropriate time when the changes to `dev` are merged back into `main` and then tagged with a release number while bumping `MAJOR` and/or `MINOR`.
```

we can revisit this point in the future if necessary, but for now, we leave the date for `v1.1.0` TBD.

what do you think @Fi3 ?

Reply:
> I'm sure we will need to make more patch releases than feature ones btw

I can understand this point, and I wonder if we can do something about it. While we already have a strategy for dealing with bugs under `SemVer 2.0.0`, IMO we should keep that for exceptional situations where finding a bug is actually a **surprise**.

With that in mind, I propose that we start our release cycle with `v1.0.0-beta.0` instead of `v1.0.0`.
As we fix bugs, we increase `v1.0.0-beta.1`, `v1.0.0-beta.2`, etc.

Please let me know your thoughts @Fi3 @pavlenex @GitGab19

Reply:
I understand your point about `v1.0.0-beta.1, v1.0.0-beta.2` but I am slightly worried on what that communicates to the public, and something as simple as a tag could drive people away from implementing and just keep them in "I'll wait a bit more stage". 

1.0.0 communicates that after 3+ years this is ready to be tested and adopted, sure there will be bugs, sure there's a LOT of work to be done to ensure stability, and probably a whole can of worms once different configurations and use-cases start being used, but... this is our opportunity to start fresh and clean, and I feel these tags would kill the momentum of excitement around something as significant as 1.0.0, so I'll really keep things as we already agreed, besides timeframe for releases seems we have a consensus. Timelines can easily be adjusted.

Reply:
Yeah now I would try to release it as `1.0.0`, without specifying a timeframe for merging `dev` into `main` and push a new release. After that, we will see how we can be more specific about timeframes, depending on external feedback and testing!

Reply:
ack https://github.com/stratum-mining/stratum/pull/766/commits/2c4cc3f71d1fc324d85086ec6e9d9396e7c9772f